components/supervisor/pkg/supervisor/supervisor.go

```diff
@@ -147,7 +148,7 @@ func Run(options ...RunOption) {
 
 	ctx, cancel := context.WithCancel(context.Background())
 	var (
-		shutdown            = make(chan struct{})
+		shutdown            = make(chan *bool, 1)
```

Comment:
`chan *bool` strikes me as really odd. Why not `chan bool`?

Reply:
To differentiate between "not set" and `false`. Is there a more "go-ish" way to do so?

Reply:
I'd make this an enum and make the three different states explicit